rafthttp/transport.go

```diff
@@ -85,17 +93,21 @@ type transport struct {
 	raft         Raft
 	serverStats  *stats.ServerStats
 	leaderStats  *stats.LeaderStats
+	v3demo       bool
 
-	mu      sync.RWMutex         // protect the term, remote and peer map
-	term    uint64               // the latest term that has been observed
-	remotes map[types.ID]*remote // remotes map that helps newly joined member to catch up
-	peers   map[types.ID]Peer    // peers map
+	mu         sync.RWMutex         // protect below fields
+	term       uint64               // the latest term that has been observed
+	remotes    map[types.ID]*remote // remotes map that helps newly joined member to catch up
+	peers      map[types.ID]Peer    // peers map
+	urlPickers map[types.ID]*urlPicker
```

Comment:
I think you still do not answer my original question or my question is not clear enough. My question is why do we need to have a picker map here in `transporter`.